From c2b7edd8b755d3b0c783f3db5d89278bc3c8b2d1 Mon Sep 17 00:00:00 2001
From: Adrian DC <radian.dc@gmail.com>
Date: Tue, 20 Mar 2018 23:48:08 +0100
Subject: [PATCH 10/10] libstagefright: Support disabling metadata with a
 property

 * Allow defining a device property called
    "debug.camcorder.disablemeta" to force
    storeMetaDataInVideoBuffers to false, thus allowing
    the video YUV fallback to work properly

 * Forward port and adaptation from codeaurora commit :

    Author: Haynes Mathew George <hgeorge@codeaurora.org>
    Date:   Wed, 29 Aug 2012 22:58:51 +0300
    libstagefright: support for disabling buffer metadata

      - Metada mode video recording is enabled by default.
      - use setprop debug.camcorder.disablemeta 1 to disable metadata mode recording.

     PS2: Disable metadata mode in timelapse mode too.
     Change-Id: I422c49c0ace0c3a3e1f4459c7e4bf29e70af763a

 * Similar to the following initial header changes:

    Author: mickybart <mickybart@pygoscelis.org>
    Date:   Fri, 9 Mar 2018 15:04:02 -0500
    Force the CameraSource to not use MetaData for HAL1
    Change-Id: I4f84025f824aee332883c8ce34d7afe8c383cba2

Change-Id: Ia5f3bd30cd24bd2d10c9c145c91d4b616f4edb44
---
 media/libstagefright/CameraSource.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/media/libstagefright/CameraSource.cpp b/media/libstagefright/CameraSource.cpp
index 36a2d3fed..ce5be98e7 100644
--- a/media/libstagefright/CameraSource.cpp
+++ b/media/libstagefright/CameraSource.cpp
@@ -232,6 +232,10 @@ CameraSource::CameraSource(
     mVideoSize.width  = -1;
     mVideoSize.height = -1;
 
+    if (property_get_bool("debug.camcorder.disablemeta", false)) {
+        storeMetaDataInVideoBuffers = false;
+    }
+
     mInitCheck = init(camera, proxy, cameraId,
                     clientName, clientUid, clientPid,
                     videoSize, frameRate,
-- 
2.17.0

